tracer/build/_build/Build.GitHub.cs

```diff
@@ -91,29 +91,23 @@ partial class Build
            .Requires(() => GitHubRepositoryName)
            .Requires(() => GitHubToken)
            .Requires(() => PullRequestNumber)
+           .Requires(() => TargetBranch)
            .Executes(async() =>
             {
-                var client = GetGitHubClient();
-
-                var pr = await client.PullRequest.Get(
-                    owner: GitHubRepositoryOwner,
-                    name: GitHubRepositoryName,
-                    number: PullRequestNumber.Value);
-
-                // Fixes an issue (ambiguous argument) when we do git diff in the Action.
-                GitTasks.Git("fetch origin master:master", logOutput: false);
+                // This assumes that we're running in a pull request, so we compare against the target branch
+                var baseCommit = GitTasks.Git($"merge-base origin/{TargetBranch} HEAD").First().Text;
 
                 // This is a dumb implementation that just show the diff
                 // We could imagine getting the whole context with -U1000 and show the differences including parent name
                 // eg now we show -oldAttribute: oldValue, but we could show -tag.oldattribute: oldvalue
-                var changes = GitTasks.Git("diff master -- tracer/test/snapshots")
+                var changes = GitTasks.Git($"diff \"{baseCommit}\" -- */*snapshots*/*.*")
                                    .Select(f => f.Text);
```

Comment:
I have an alternative suggestion to the "exclude count". The intention there (IIUC) is to not flood the PR with comments when new snapshots have been added? In which case, we could limit the analysis to _only_ snapshots that have been modified by adding a `--diff-filter`:

```suggestion
                var changes = GitTasks.Git($"diff --diff-filter=M \"{baseCommit}\" -- */*snapshots*/*.*")
                                   .Select(f => f.Text);
```

This excludes renamed/deleted/added snapshots, so I think you could probably remove the following `nbSnapshotsModified` logic entirely?

Reply:
Thanks for the suggestion, it's way better. Done in 4a4cb032df0f3a8c9b72787dea79e2b1d07fd682